From f0c609e9855a3b3c39eb0fac96e712a63e89e6d3 Mon Sep 17 00:00:00 2001
From: Cristian Cozzolino <cristian_ci@protonmail.com>
Date: Sun, 18 Oct 2020 19:13:48 +0200
Subject: [PATCH 149/210] arm64: dts: qcom: msm8916-lg-m216: Add sound and
 modem (v4)

Signed-off-by: Cristian Cozzolino <cristian_ci@protonmail.com>

MODEM:
v2: Setup memshare / add GPS memory
v3: wcnss/venus no longer need to be relocated
v4: Combine sound and modem commits and drop extra -modem.dts
    The no-modem variant is now patched dynamically by lk2nd instead.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
---
 arch/arm64/boot/dts/qcom/msm8916-lg-m216.dts | 30 ++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-lg-m216.dts b/arch/arm64/boot/dts/qcom/msm8916-lg-m216.dts
index 263604253e2..65f438d7f57 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-lg-m216.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-lg-m216.dts
@@ -3,6 +3,8 @@
 /dts-v1/;
 
 #include "msm8916-pm8916.dtsi"
+#include "msm8916-modem.dtsi"
+
 #include <dt-bindings/gpio/gpio.h>
 
 / {
@@ -18,6 +20,18 @@ chosen {
 		stdout-path = "serial0";
 	};
 
+	reserved-memory {
+		mpss_mem: mpss@86800000 {
+			reg = <0x0 0x86800000 0x0 0x4a00000>;
+			no-map;
+		};
+
+		gps_mem: gps@8b200000 {
+			reg = <0x0 0x8b200000 0x0 0x200000>;
+			no-map;
+		};
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 
@@ -150,6 +164,14 @@ &sdhc_2 {
 	cd-gpios = <&msmgpio 38 GPIO_ACTIVE_LOW>;
 };
 
+&sound {
+	status = "okay";
+	audio-routing =
+		"AMIC1", "MIC BIAS External1",
+		"AMIC2", "MIC BIAS Internal2",
+		"AMIC3", "MIC BIAS External1";
+};
+
 &usb {
 	status = "okay";
 	dr_mode = "peripheral";
@@ -160,6 +182,14 @@ &usb_hs_phy {
 	extcon = <&pm8916_usbin>;
 };
 
+&wcd_codec {
+	qcom,micbias1-ext-cap;
+	qcom,micbias-lvl = <2800>;
+	qcom,mbhc-vthreshold-low = <75 100 120 180 500>;
+	qcom,mbhc-vthreshold-high = <75 100 120 180 500>;
+	qcom,hphl-jack-type-normally-open;
+};
+
 &smd_rpm_regulators {
 	vdd_l1_l2_l3-supply = <&pm8916_s3>;
 	vdd_l4_l5_l6-supply = <&pm8916_s4>;
-- 
2.34.1

